source/ui/util/AMChooseDataFolderDialog.cpp

```diff
@@ -46,12 +48,19 @@ bool AMChooseDataFolderDialog::getDataFolder(const QString &localRootDirectory,
 				AMUserSettings::save(true);
 			}
 
-			else {
-
-				AMUserSettings::userDataFolder = QString("%1/%2/%3/userData/").arg(localRootDirectory).arg(dataDirectory).arg(dialogInput);
-				AMUserSettings::remoteDataFolder = QString("%1/%2/%3/userData/").arg(remoteRootDirectory).arg(dataDirectory).arg(dialogInput);
-				AMUserSettings::save();
-			}
+                        else if(successfulFirstTimeStartup){
+
+                                AMUserSettings::userDataFolder = QString("%1/%2/%3/userData/").arg(localRootDirectory).arg(dataDirectory).arg(dialogInput);
+                                AMUserSettings::remoteDataFolder = QString("%1/%2/%3/userData/").arg(remoteRootDirectory).arg(dataDirectory).arg(dialogInput);
+                                AMUserSettings::save();
+                        }
+                        else{
+                                //If this is not the first startup and first startup was not successful:
```

Comment:
The main bug was caused by bad .ini configurations, so I only checked folders referenced in the config file. Ideally `extraDataDirectory` entries are always made at Ln43 before we reach the cases for a good or bad startup wizard run. `extraDataDirectory` entries were not validated before, maybe they should be? I'm more then happy to change it but that was my reasoning.